src/ProductConstructionService/ProductConstructionService.SubscriptionTriggerer/SubscriptionTriggerer.cs

```diff
@@ -0,0 +1,94 @@
+        ILogger<SubscriptionTriggerer> logger,
+        BuildAssetRegistryContext context,
+        IBasicBarClient barClient,
+        OperationManager operations,
+        QueueClient queueClient)
+    {
+        _operations = operations;
+        _logger = logger;
+        _context = context;
+        _barClient = barClient;
+        _queueClient = queueClient;
+    }
+
+    public async Task CheckSubscriptionsAsync(UpdateFrequency targetUpdateFrequency)
+    {
+        using (_operations.BeginOperation($"Updating {targetUpdateFrequency} subscriptions"))
+        {
+            var enabledSubscriptionsWithTargetFrequency = (await _context.Subscriptions
+                    .Where(s => s.Enabled)
+                    .ToListAsync())
+                    .Where(s => s.PolicyObject?.UpdateFrequency == targetUpdateFrequency);
+
+            int subscriptionsUpdated = 0;
+            foreach (var subscription in enabledSubscriptionsWithTargetFrequency)
+            {
+                Subscription? subscriptionWithBuilds = await _context.Subscriptions
+                    .Where(s => s.Id == subscription.Id)
+                    .Include(s => s.Channel)
+                    .ThenInclude(c => c.BuildChannels)
+                    .ThenInclude(bc => bc.Build)
+                    .FirstOrDefaultAsync();
+
+                if (subscriptionWithBuilds == null)
+                {
+                    _logger.LogWarning("Subscription {subscriptionId} was not found in the BAR. Not applying updates", subscription.Id.ToString());
+                    continue;
+                }
+
+                Build? latestBuildInTargetChannel = subscriptionWithBuilds.Channel.BuildChannels.Select(bc => bc.Build)
+                    .Where(b => (subscription.SourceRepository == b.GitHubRepository || subscription.SourceRepository == b.AzureDevOpsRepository))
+                    .OrderByDescending(b => b.DateProduced)
+                    .FirstOrDefault();
+
+                bool isThereAnUnappliedBuildInTargetChannel = latestBuildInTargetChannel != null &&
+                    (subscription.LastAppliedBuild == null || subscription.LastAppliedBuildId != latestBuildInTargetChannel.Id);
+
+                if (isThereAnUnappliedBuildInTargetChannel && latestBuildInTargetChannel != null)
+                {
+                    _logger.LogInformation("Will update {subscriptionId} to build {latestBuildInTargetChannelId}", subscription.Id, latestBuildInTargetChannel.Id);
+                    UpdateSubscriptionAsync(subscription.Id, latestBuildInTargetChannel.Id);
+                    subscriptionsUpdated++;
+                }
+            }
+
+            _logger.LogInformation("Updated '{SubscriptionsUpdated}' '{targetUpdateFrequency}' subscriptions", subscriptionsUpdated, targetUpdateFrequency.ToString());
+        }
+    }
+
+    private void UpdateSubscriptionAsync(Guid subscriptionId, int buildId)
+    {
+        using (_operations.BeginOperation(
```

Comment:
I'd remove the operation, yeah

Reply:
removed the operation thing from the project altogether